tests/unit/tuner/callback/test_early_stopping.py

```diff
@@ -0,0 +1,145 @@
+import pytest
```

Comment:
General comment: you do not need to use the `tuner.fit` function here at all, just create a tuner and then call the callback method with that tuner as argument directly. Calling the `.fit` function should only be done in an integration test - but designing one for this might not be so easy, and it might not be needed.

And current tests do not actually test the early stopping functionality - i guess because you can not so easily create a model and data that would give you the desired behavior when tuning. that's where calling callback methods directly is useful

Reply:
I agree I changed all the tests and removed .fit